corehq/apps/registration/utils.py

```diff
@@ -131,17 +132,38 @@ def request_new_domain(request, form, is_new_user=True):
 
     if is_new_user:
         dom_req.save()
-        send_domain_registration_email(request.user.email,
-                                       dom_req.domain,
-                                       dom_req.activation_guid,
-                                       request.user.get_full_name(),
-                                       request.user.first_name)
+        if settings.IS_SAAS_ENVIRONMENT:
+            load_appcues_template_apps.apply_async(
+                (new_domain.name, current_user.username),
+                link=send_domain_registration_email(
+                    request.user.email,
+                    dom_req.domain,
+                    dom_req.activation_guid,
+                    request.user.get_full_name(),
+                    request.user.first_name
+                )
+            )
+        else:
+            send_domain_registration_email(request.user.email,
+                                           dom_req.domain,
+                                           dom_req.activation_guid,
+                                           request.user.get_full_name(),
+                                           request.user.first_name)
     send_new_request_update_email(request.user, get_ip(request), new_domain.name, is_new_user=is_new_user)
 
     send_hubspot_form(HUBSPOT_CREATED_NEW_PROJECT_SPACE_FORM_ID, request)
     return new_domain.name
 
 
+APPCUES_TEMPLATE_SLUGS = ['appcues']  # TODO: Add agg and wash
+
+@task(serializer='pickle', queue='background_queue')
```

Comment:
Is the background queue definitely the right place for these? The other alternative I could think of is to put them on the email queue, since they're blocking an email. Wondering if putting them on the email queue would save them from being delayed when the background queue gets clogged up. Although, of course, they're not emails so perhaps that's confusing. I don't have strong philosophical opinions on celery or a great understanding of exactly what happens when it gets backed up, just wondering if you've considered putting them on a different queue.

Reply:
I actually like that idea. I didn't consider it because I don't know if we have any formal or informal rule about limiting celery queues to similar things. As far as I can tell, these are the queues that we have:
background_queue
email_queue
sms_queue
some icds queues
some reminder queues
the celery_periodic queue
and the main celery queue

I think it would make the most sense to put this task on the email or background queues. As a team, do we feel strongly about mixing tasks and queues like that?

Reply:
I don't know - @proteusvacuum @emord @calellowitz  are the people I'd guess have feelings about this.

Reply:
I don't have strong feelings about this - I think the background queue seems fine for this.

Reply:
Generally speaking, I think this belongs in the background_queue. In theory the email_queue is just for sending emails and the background queue or main queue (celery) is for this sort of task, but we aren't necessarily great at being consistent, so those are pretty weak opinions. I think the background queue is supposed to be for things that we do not mind if they are backed up and the main queue isn't so this might fit better in the main queue if we have strong opinions about how soon this email is delivered. However, I do not know which of the two gets more behind in practice, I just know that we may not take much action if the background queue gets behind and we would if the main one does. 

On a broader note, do we expect this task to take a while? Celery does have a habit of getting backed up and it would be unfortunate if our confirmation emails got delayed because of that. If this can be synchronous it might best. The task itself looks highly cacheable so a heavily cached version of that called synchronously may be a good way to prevent issues.